src/main/java/org/broadinstitute/hellbender/tools/funcotator/genelistoutput/GeneListOutputRenderer.java

```diff
@@ -0,0 +1,255 @@
+    static final String INVALID_VARIANT_CONTEXT_ERR_MSG = "  Variant context does not represent a copy number segment";
+    @VisibleForTesting
+    static final String INCORRECT_NUM_TRANSCRIPT_ERR_MSG = "  Need exactly one transcript ID";
+    @VisibleForTesting
+    static final String INVALID_TRANSCRIPT_ID_ERR_MSG = "  Invalid transcript ID seen";
+
+    final private SortedMap<Pair<String,String>, Pair<VariantContext, FuncotationMap>> geneExonToVariantFuncotationMap;
+    final private SimpleTsvOutputRenderer simpleTsvOutputRenderer;
+    final private Pattern gencodeGenesFieldPattern;
+    final private Pattern gencodeStartExonFieldPattern;
+    final private Pattern gencodeStartGeneFieldPattern;
+    final private Pattern gencodeEndGeneFieldPattern;
+    final private Pattern gencodeEndExonFieldPattern;
+
+    public GeneListOutputRenderer(final Path outputFilePath,
+                                  final LinkedHashMap<String, String> unaccountedForDefaultAnnotations,
+                                  final LinkedHashMap<String, String> unaccountedForOverrideAnnotations,
+                                  final Set<String> excludedOutputFields, final String toolVersion){
+        super(toolVersion);
+        Utils.nonNull(unaccountedForDefaultAnnotations);
+        Utils.nonNull(unaccountedForOverrideAnnotations);
+        Utils.nonNull(excludedOutputFields);
+        Utils.nonNull(toolVersion);
+        Utils.nonNull(outputFilePath);
+        IOUtil.assertFileIsWritable(outputFilePath.toFile());
+
+        geneExonToVariantFuncotationMap = new TreeMap<>(new PairedSimpleNaturalComparator<>());
+
+        // When initializing the simpleoutput renderer, make sure that extra funcotation fields not explicitly in the
+        //  configuration are not written.
+        simpleTsvOutputRenderer = SimpleTsvOutputRenderer.createFromResource(outputFilePath,
+            unaccountedForDefaultAnnotations, unaccountedForOverrideAnnotations, excludedOutputFields,
+            Paths.get(CONFIG_RESOURCE), toolVersion, false);
+
+        // Cache patterns to determine which field
+        gencodeGenesFieldPattern = Pattern.compile(GENCODE_GENES_FIELD_PATTERN_STRING);
+        gencodeStartGeneFieldPattern = Pattern.compile(GENCODE_START_GENE_FIELD_PATTERN_STRING);
+        gencodeStartExonFieldPattern = Pattern.compile(GENCODE_START_EXON_FIELD_PATTERN_STRING);
+        gencodeEndGeneFieldPattern = Pattern.compile(GENCODE_END_GENE_FIELD_PATTERN_STRING);
+        gencodeEndExonFieldPattern = Pattern.compile(GENCODE_END_EXON_FIELD_PATTERN_STRING);
+    }
+
+    /**
+     * Since we have to wait until all segments are processed, the output renderer will do nothing until closed.
+     */
+    @Override
+    public void close() {
+
+        // Flush the map.
+        flushAllGenes();
+
+        simpleTsvOutputRenderer.close();
+    }
+
+    /** Since this output renderer needs to see all of the input before it can render properly, it will not actually
```

Comment:
As long as the number of genes does not increase dramatically (or we get a TON of small segments), we should be okay for RAM, since we store a map with gene as keys.

No action.